wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/JvmManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.system.server;
+
+import com.google.common.io.Files;
+import com.sun.management.HotSpotDiagnosticMXBean;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipOutputStream;
+import javax.inject.Singleton;
+import org.eclipse.che.commons.lang.IoUtil;
+import org.eclipse.che.commons.lang.ZipUtils;
+
+@Singleton
+public class JvmManager {
+
+  public static final DateFormat MILLIS_FORMAT = new SimpleDateFormat("mm:ss:SSS");
+  private final ThreadMXBean threadMxBean;
+  private final HotSpotDiagnosticMXBean hotSpotMxBean;
+
+  public JvmManager() {
+    threadMxBean = ManagementFactory.getThreadMXBean();
+    hotSpotMxBean = ManagementFactory.getPlatformMXBean(HotSpotDiagnosticMXBean.class);
+  }
+
+  public void writeThreadDump(OutputStream outputStream) throws IOException {
+    OutputStreamWriter writer = new OutputStreamWriter(outputStream);
+    ThreadInfo[] threadInfos =
+        threadMxBean.getThreadInfo(threadMxBean.getAllThreadIds(), Integer.MAX_VALUE);
+    Map<Long, ThreadInfo> threadInfoMap = new HashMap<>();
+    for (ThreadInfo threadInfo : threadInfos) {
+      threadInfoMap.put(threadInfo.getThreadId(), threadInfo);
+    }
+
+    try {
```

Comment:
Seems it's try used only to close the writer in `finally` statement. But some runtime exception can happen above and then the writer won't be closed if I'm not mistaken. Please consider moving the section above in try as well.